### PR TITLE
HPCC-20292 Port 0 causes issues when starting ESP

### PR DESCRIPTION
**Pull request by mayx**
- Don't load a binding if port is configured to be 0

Signed-off-by: mayx <yanrui.ma@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by mayx** [comment]
@afishbeck  please review.

**Comment by mayx** [comment]
@rpastrana please review...

File: esp/platform/espcfg.cpp [hunk 1]
```diff
@@ -263,34 +263,36 @@ CEspConfig::CEspConfig(IProperties* inputs, IPropertyTree* envpt, IPropertyTree*
...
-                    
-                    StringBuffer addr;
-                    ptree->getProp("@netAddress", addr);
-                    if(strcmp(addr.str(), ".") == 0)
-                    {
-                        bcfg->address.append("0.0.0.0");
-                    }
+                    bcfg->port = ptree->getPropInt("@port", 0);
+                    if (bcfg->port == 0)
+                        DBGLOG("Binding %s is configured with port 0, do not load it.", bcfg->name.str());
```
Comment (rpastrana) [inline esp/platform/espcfg.cpp]: Unless it is 100% assured that bcfg will always have a port attribute, and we never default to 0, this message could be misleading. Also, we could say  ... is configured with port 0, and will not be loaded.

File: esp/platform/espcfg.cpp [hunk 2]
```diff
@@ -263,34 +263,36 @@ CEspConfig::CEspConfig(IProperties* inputs, IPropertyTree* envpt, IPropertyTree*
                 ptree = &pt_iter->query();
                 if (ptree)
                 {
-                    binding_cfg *bcfg = new binding_cfg;
-                    
+                    OwnedPtr<binding_cfg> bcfg(new binding_cfg);
```
Comment (rpastrana) [inline esp/platform/espcfg.cpp]: the if clause performed more actions before this change, please confirm the logic change doesn't introduce any side effects other then the stated port 0 binding change.

**Review by rpastrana — COMMENTED** [review]
@mayx a couple of comments

**Comment by mayx** [comment]
@rpastrana wording change uploaded, please review again...

**Review by rpastrana — APPROVED** [review]
@mayx approved

**Pull request merged**